src/libexpr/lexer.l

```diff
@@ -64,8 +65,9 @@ static Expr * unescapeStr(SymbolTable & symbols, const char * s, size_t length)
             else if (c == 'r') t += '\r';
             else if (c == 't') t += '\t';
             else {
-                assert(c == '"' || c == '\\' || c == '$' || c == '\n');
-                t += c;
+              if (!(c == '"' || c == '\\' || c == '$' || c == '\n'))
+                std::cerr << "meaningful escape sequence \"\\" << c << "\" at " << loc->last_line << ":" << loc->last_column << std::endl;
```

Comment:
Not sure if writing to `std::cerr` is appropriate here, but doesn't look like a logger is in scope either.  Not sure it's warranted but perhaps warnings can be pushed into a deferred list and processed by the caller at the end?

Message seems inverted-- it says "meaningful escape sequence" but only is triggered when a character is not one of the supported ("meaningful"?) ones.

Reply:
writer/collector for warnings might be needed, yes.
there could be more warnings, on unused variables or needless `rec` keyword or empty `let in`.